### ISSUE TITLE
Several suggestions to improve 'words'

### ISSUE THREAD

**Issue by learnbyexample** [30-seconds/30-seconds-of-python#280]
Link: https://www.30secondsofcode.org/python/s/words. Existing snippet:

```python
def words(str, pattern = '[^a-zA-Z-]+'):
    return list(filter(None, re.split(pattern, str)))
```

1. Use `re.findall` instead of `re.split` and `filter`. This will also change to positive character class instead of negated one.

```python
def words(str, pattern = '[a-zA-Z-]+'):
    return re.findall(pattern, str)
```

2. `str` is a built-in function and raw-strings are preferred to define regular expression patterns

```python
def words(ipstr, pattern = r'[a-zA-Z-]+'):
    return re.findall(pattern, ipstr)
```

3. The description isn't precise. Hyphen is missing from "defaults to non-alpha as a regexp".

```python
>>> def words(ipstr, pattern = r'[a-zA-Z-]+'):
...     return re.findall(pattern, ipstr)
... 
>>> words('good -123 -baz in-built foo-')
['good', '-', '-baz', 'in-built', 'foo-']
```

4. You could add another example to show use of second argument, as well as illustrate how word boundaries can be useful. Note the use of raw-string, with normal strings, you'll need `'\\b[a-zA-Z-]+\\b'`

```python
>>> words('good -123 -baz in-built foo-', r'\b[a-zA-Z-]+\b')
['good', 'baz', 'in-built', 'foo']
```

**Comment by Trinityyi** [comment]
1. Yes, this seems like a valid improvement.
2. Correct. `s` is more appropriate as per the guidelines and other snippet conventions, instead of `ipstr`.
3. True, it could be` defaults to non-alpha and hyphens as regexp`. (or `alphanumeric and hypens` if the regexp is swapped.
4. Definitely worth doing that.

Overall, the suggestions are all solid. Could you open a PR for this, if you have the time? Remember to tag `Fixes #280` in the description.

**Comment by learnbyexample** [comment]
>Could you open a PR for this

https://github.com/30-seconds/30-seconds-of-python/blob/master/CONTRIBUTING.md says `npm` is needed, which I'm not willing to install/learn. I do not mind someone else who is familiar with this project to make the changes.
